lib/worldwide/address.rb

```diff
@@ -465,6 +465,35 @@ def strip_extra_japanese_chars(line:, excluded_fields:)
       end
     end
 
+    def script_from_string(str)
+      script = "default"
+      scripts = region.combined_address_format.except("default").keys
+      return script if scripts.empty?
+
+      detected_scripts = Worldwide::Scripts.identify(text: str).map(&:to_s)
+      if detected_scripts.length == 1 && scripts.include?(detected_scripts.first)
+        script = detected_scripts.first
+      end
+      script
+    end
+
+    def script_from_field(field)
+      scripts = region.combined_address_format.except("default").keys
+      unless scripts.empty?
+        script = region.combined_address_format.except("default").keys.find do |script|
```

Comment:
We want to identify the script(s) used in the relevant address fields, but at this point in the code, we don't know what the additional address fields are - we need the script key to know what the relevant additional address fields are for concatenation. So we iterate over each script